src/index.ts

```diff
@@ -1,112 +1,205 @@
+      },
+    });
+  }
+
+  /**
+   * Track an event
+   */
   static async track(...[event, category, props]: Event) {
-    if (!Segment.loopName || !Segment.userId) {
-      throw new Error('Analytics not initialized');
+    await Segment.postRequest('track', {
+      event,
+      properties: {
+        ...props,
+        whisper_name: Segment.currentPage,
+        category,
+        label: props
+          ? Object.entries(props)
+              .map(([k, v]) => `[${k}: ${v}]`)
+              .join(' ')
+          : undefined,
+      },
+    });
+  }
+
+  static async trackComponentClicked(componentType: whisper.WhisperComponentType) {
+    await Segment.track(EventName.ComponentClicked, EventCategory.ClickEvents, {
+      component_type: componentType,
+    });
+  }
+
+  static async trackComponentCopied(componentType: whisper.WhisperComponentType) {
+    await Segment.track(EventName.ComponentCopied, EventCategory.ClickEvents, {
+      component_type: componentType,
+    });
+  }
+
+  static async trackWhisperClosed() {
+    await Segment.track(EventName.WhisperClosed, EventCategory.Whispers);
+  }
+
+  static async trackWhisperDisplayed(whisperName: string, whisperUpdated: boolean) {
+    await Segment.page(whisperName, whisperUpdated);
+  }
+
+  // To be removed if considered redundant
+  static async trackWhisperTriggered(whisperName: string, triggeredFrom: string) {
+    await Segment.track(EventName.WhisperTriggered, EventCategory.Whispers, {
+      whisper_name: whisperName,
+      triggered_from: triggeredFrom,
+    });
+  }
+
+  /**
+   * Sends the HTTP request to Segment based on given endpoint and body args
+   *
+   * Should not be called directly, use identify, page, or track instead
+   */
+  private static async postRequest(...[endpoint, body]: SegmentRequest) {
+    if (!Segment.initialized) {
+      throw new Error('Segment not initialized');
     }
-    const properties = {
-      loop_name: Segment.loopName,
-      ...props,
-    };
+    const { loopName, userId, writeKey } = Segment.config;
+
     const request: network.HTTPRequest = {
-      url: `${Segment.BASE_URL}/track`,
+      url: `${Segment.BASE_URL}/${endpoint}`,
       method: 'POST',
       headers: {
-        Authorization: [`Basic ${btoa(`${Segment.writeKey}:`)}`],
+        Authorization: [`Basic ${btoa(`${writeKey}:`)}`],
         'Content-Type': ['application/json'],
       },
       body: JSON.stringify({
-        userId: Segment.userId,
-        event,
+        ...body,
+        userId,
         properties: {
-          ...properties,
-          category,
-          label: properties
-            ? Object.entries(properties)
-                .map(([k, v]) => `[${k}: ${v}]`)
-                .join(' ')
-            : undefined,
+          ...body.properties,
+          loop_name: loopName,
         },
       }),
     };
-    const { statusCode, body } = await network.httpRequest(request);
-    if (statusCode >= 200 && statusCode <= 300) {
```

Comment:
Is there a reason you removed the status code checking. Or is it something you plan on adding back with the refactor. I would lean towards adding it back if thats not planned.

Reply:
I removed it because the loop is just firing and forgetting the analytics calls, so I wouldn't think we necessarily care whether or not the call was successful. There's no response value that we need and we don't want the loop to fail or slow down because of analytics

Reply:
Makes sense to me. I think the only thing is I would try to let the user know that segment (or GA) failed its first call and most likely all tracking calls wont make it or something of the sort. Like if the identify call fails (which looks like to be the first call to segment) set initialized to false

Reply:
Went with the direction of logging errors if the class isn't initialized as well as logging errors if the network calls fail. Decided not to set initialized to false if the first call fails because the developer did successfully initialize the config for the class, the network call just failed